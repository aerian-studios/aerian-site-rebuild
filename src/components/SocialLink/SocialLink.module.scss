@import "../../scss/variables";

.item {
    list-style: none;
    display: inline-block;
    margin-right: calc(var(--s-1) * 4);
    font-size: var(--t-h2);
    a {
        font-size: 1.6em;
    }
    svg {
        fill: #d8d8d8;
    }
    &:last-child {
        margin-right: 0;
    }
}
