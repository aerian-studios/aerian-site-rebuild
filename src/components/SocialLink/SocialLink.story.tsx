import * as React from "react";

import { withInfo } from "@storybook/addon-info";
import { storiesOf } from "@storybook/react";

import { SocialLink } from "./index";

storiesOf("SocialLink", module).add(
    "Default",
    withInfo({ inline: true })(() => (
        <SocialLink
            className="item"
            iconName="facebook-f"
            url="https://www.facebook.com"
        />
    ))
);
