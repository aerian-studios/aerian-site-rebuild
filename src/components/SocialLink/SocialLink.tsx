import * as React from "react";

import { IconName } from "@fortawesome/fontawesome-svg-core";

import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

import * as styles from "./SocialLink.module.scss";

interface Props {
    style?: React.CSSProperties;
    className?: string;
    iconName: IconName;
    url: string;
}

export const SocialLink: React.SFC<Props> = ({
    style,
    className,
    iconName,
    url
}) => (
    <span className={[styles.item, className].join(" ")} style={style}>
        <a href={url} rel="nofollow noreferer">
            <FontAwesomeIcon icon={["fab", iconName]} />
        </a>
    </span>
);
export default SocialLink;
