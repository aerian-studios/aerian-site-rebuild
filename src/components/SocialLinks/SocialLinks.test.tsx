import * as React from "react";

import * as renderer from "react-test-renderer";

import { faFacebook } from "@fortawesome/free-brands-svg-icons/faFacebook";
import { faLinkedin } from "@fortawesome/free-brands-svg-icons/faLinkedin";
import { faTwitter } from "@fortawesome/free-brands-svg-icons/faTwitter";
import { faVimeoV } from "@fortawesome/free-brands-svg-icons/faVimeoV";
import { SocialLink } from "../SocialLink";
import { SocialLinks } from "./index";

describe("SocialLinks", () => {
    it("renders correctly", () => {
        const tree = renderer
            .create(
                <SocialLinks>
                    <SocialLink
                        className="item"
                        icon={faFacebook}
                        url="https://www.facebook.com/aerianstudios/"
                    />
                    <SocialLink
                        className="item"
                        icon={faTwitter}
                        url="https://twitter.com/aerianstudios"
                    />
                    <SocialLink
                        className="item"
                        icon={faLinkedin}
                        url="https://www.linkedin.com/company/aerian-studios"
                    />
                    <SocialLink
                        className="item"
                        icon={faVimeoV}
                        url="https://vimeo.com/aerianstudios"
                    />
                </SocialLinks>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
