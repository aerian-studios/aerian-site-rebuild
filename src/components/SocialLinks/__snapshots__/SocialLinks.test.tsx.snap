// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialLinks renders correctly 1`] = `
<div
  className=" "
>
  <span
    className="item"
  >
    <a
      aria-label="Visit Aerian studios on facebook"
      href="https://www.facebook.com/aerianstudios/"
      rel="nofollow noreferer"
    >
      <i
        className="fa"
      />
    </a>
  </span>
  <span
    className="item"
  >
    <a
      aria-label="Visit Aerian studios on twitter"
      href="https://twitter.com/aerianstudios"
      rel="nofollow noreferer"
    >
      <i
        className="fa"
      />
    </a>
  </span>
  <span
    className="item"
  >
    <a
      aria-label="Visit Aerian studios on linkedin"
      href="https://www.linkedin.com/company/aerian-studios"
      rel="nofollow noreferer"
    >
      <i
        className="fa"
      />
    </a>
  </span>
  <span
    className="item"
  >
    <a
      aria-label="Visit Aerian studios on vimeo-v"
      href="https://vimeo.com/aerianstudios"
      rel="nofollow noreferer"
    >
      <i
        className="fa"
      />
    </a>
  </span>
</div>
`;
